Update docs for instantiating GE (#3985)
great-expectations/great_expectations: docs/guides/setup/configuring_data_contexts/how_to_instantiate_a_data_context_without_a_yml_file.md

>     The DataContextConfig holds all of the associated configuration parameters to build a DataContext. There are defaults set for you to minimize configuration in typical cases, but please note that every parameter is configurable and all defaults are overridable. Also note that DatasourceConfig also has defaults which can be overridden.

Reviewer: ```suggestion
    The `DataContextConfig` holds all of the associated configuration parameters to build a Data Context. There are defaults set for you to minimize configuration in typical cases, but please note that every parameter is configurable and all defaults are overridable. Also note that `DatasourceConfig` also has defaults which can be overridden.
```

Reviewer: Anything outside of a header line or code block that refers to a class, method, or parameter should be indicated with back ticks.  If you are referring to something conceptually, use the technical term. (e.g. `DataContext` if you are referring to an existing object they are working with, Data Context if you are talking about it more generally.)

data_context_config = DataContextConfig(
    datasources={
        "sql_warehouse": DatasourceConfig(
            class_name="Datasource",
            execution_engine={
                "class_name": "SqlAlchemyExecutionEngine",
                "credentials": {
                    "drivername": "postgresql+psycopg2",
                    "host": "localhost",
                    "port": "5432",
                    "username": "postgres",
                    "password": "postgres",
                    "database": "postgres",
                },
            },
            data_connectors={
                "default_runtime_data_connector_name": {
                    "class_name": "RuntimeDataConnector",
                    "batch_identifiers": ["default_identifier_name"],
                },
                "default_inferred_data_connector_name": {
                    "class_name": "InferredAssetSqlDataConnector",
                    "name": "whole_table",
                },
            }
        )
    },
    store_backend_defaults=S3StoreBackendDefaults(default_bucket_name="my_default_bucket"),
    )
```

> 

Reviewer: Your code blocks need to be reviewed for formatting.  As they stand now they are causing these non-code sections to be wrapped in a code block in the netlify preview, as well.

Reviewer: For reference: These blocks were rendered as code blocks because they were indented in the markdown.  Removing the excess indent reverted them to standard text.

```python
from great_expectations.data_context.types.base import DataContextConfig, DatasourceConfig, FilesystemStoreBackendDefaults

data_context_config = DataContextConfig(
    datasources={
        "pandas": DatasourceConfig(
            class_name="Datasource",
            execution_engine={
                "class_name": "PandasExecutionEngine"
            },
            data_connectors={
                "tripdata_monthly_configured": {
                    "class_name": "ConfiguredAssetFilesystemDataConnector",
                    "base_directory": "/path/to/trip_data",
                    "assets": {
                        "yellow": {
                            "pattern": r"yellow_tripdata_(\d{4})-(\d{2})\.csv$",
                            "group_names": ["year", "month"],
                        }
                    },
                }
            },
        )
    },
    store_backend_defaults=FilesystemStoreBackendDefaults(root_directory="/path/to/store/location"),
)
```

>     The following example shows a Data Context configuration with an SQLAlchemy datasource and two GCS buckets for metadata stores, using some custom and some default prefixes. Note that you can still substitute environment variables as in the YAML based configuration to keep sensitive credentials out of your code. ``default_bucket_name``, ``default_project_name`` sets the default value for all stores that are not specified individually.

Reviewer: ```suggestion
    The following example shows a Data Context configuration with an SQLAlchemy datasource and two GCS buckets for metadata Stores, using some custom and some default prefixes. Note that you can still substitute environment variables as in the YAML based configuration to keep sensitive credentials out of your code. `default_bucket_name`, `default_project_name` sets the default value for all stores that are not specified individually.
```

Reviewer: Single back ticks suffice for denoting code elements.  Stores are a technical term, should be title cased accordingly.  (Note: Previously "Metadata Stores" was the technical term, but it has been shortened to "Stores" in a previous review leaving the format "<Type> Stores" to specifically denote a subtype of Store in order to prevent "Metadata Stores" from being mistaken as a subtype rather than the encompassing category.)

>     The resulting DataContextConfig from the following example creates an Expectations store and Data Docs using the ``my_default_bucket`` and ``my_default_project`` parameters since their bucket and project is not specified explicitly. The validations store is created using the explicitly specified ``my_validations_bucket`` and ``my_validations_project``. Further, the prefixes are set for the Expectations store and validations store, while Data Docs use the default ``data_docs`` prefix.

Reviewer: ```suggestion
    The resulting `DataContextConfig` from the following example creates an Expectations Store and Data Docs using the `my_default_bucket` and `my_default_project` parameters since their bucket and project is not specified explicitly. The Validations Store is created using the explicitly specified `my_validations_bucket` and `my_validations_project`. Further, the prefixes are set for the Expectations Store and Validations Store, while Data Docs use the default `data_docs` prefix.
```

Reviewer: Again, single back ticks suffice.  Also, subtypes of Store are considered a technical term in and of themselves.  So "Expectations Store" is a single technical term and both words should be titlecased, as should "Validations Store"

Reviewer: The above two portions of text are also being shown as a code block; double check how those are being denoted in your source file.

                    "drivername": "postgresql+psycopg2",
                    "host": "localhost",
                    "port": "5432",
                    "username": "postgres",
                    "password": "postgres",
                    "database": "postgres",
                },
            },
            data_connectors={
                "default_runtime_data_connector_name": {
                    "class_name": "RuntimeDataConnector",
                    "batch_identifiers": ["default_identifier_name"],
                },
                "default_inferred_data_connector_name": {
                    "class_name": "InferredAssetSqlDataConnector",
                    "name": "whole_table",
                },
            }
        )
    },
    store_backend_defaults=GCSStoreBackendDefaults(
        default_bucket_name="my_default_bucket",
        default_project_name="my_default_project",
        validations_store_bucket_name="my_validations_bucket",
        validations_store_project_name="my_validations_project",
        validations_store_prefix="my_validations_store_prefix",
        expectations_store_prefix="my_expectations_store_prefix",
    ),
)
```

>     The following example sets overrides for many of the parameters available to you when creating a DataContextConfig and a Datasource.

Reviewer: Another block of text that should not be being shown as a code block, but is.

>     The following example sets overrides for many of the parameters available to you when creating a DataContextConfig and a Datasource.

Reviewer: ```suggestion
    The following example sets overrides for many of the parameters available to you when creating a `DataContextConfig` and a Datasource.
```

        "s3_site": {
            "class_name": "SiteBuilder",
            "store_backend": {
                "class_name": "TupleS3StoreBackend",
                "bucket":  "my_data_docs_bucket",
                "prefix":  "my_optional_data_docs_prefix",
            },
            "site_index_builder": {
                "class_name": "DefaultSiteIndexBuilder",
                "show_cta_footer": True,
            },
        }
    },
    validation_operators={
        "action_list_operator": {
            "class_name": "ActionListValidationOperator",
            "action_list": [
                {
                    "name": "store_validation_result",
                    "action": {"class_name": "StoreValidationResultAction"},
                },
                {
                    "name": "store_evaluation_params",
                    "action": {"class_name": "StoreEvaluationParametersAction"},
                },
                {
                    "name": "update_data_docs",
                    "action": {"class_name": "UpdateDataDocsAction"},
                },
            ],
        }
    },
    anonymous_usage_statistics={
      "enabled": True
    }
)
```

> 2. **Pass this DataContextConfig as a project_config to BaseDataContext**

Reviewer: ```suggestion
### 2. Pass this DataContextConfig as a project_config to BaseDataContext
```

Reviewer: Rather than have these denoted as bold text, it is more appropriate to have them marked as H3 sub sections.  The H1 sub section "Steps" above should be denoted as an H2 sub section (H1 is reserved for the page title).

```python
context = BaseDataContext(project_config=data_context_config)
```

> 3. **Use this BaseDataContext instance as your DataContext**

Reviewer: ```suggestion
### 3. Use this BaseDataContext instance as your DataContext
```

Reviewer: As with Section 2 above, this should be a H3 subsection rather than bold text.  Note that Section 1 should receive the same treatment, and the Additional Resources section at the bottom of the page should be an H2 subsection.
